### PR TITLE
[Wallet] Support push notifications with an "open url" semantic

### LINKED ISSUE #5325: Allow push notifications to redirect the user to a URL (for gathering feedback)
### Expected Behavior

Receiving a push notification with a URL in the data payload (field name to be determined) would redirect the user to that URL, opening the browser.

### Current Behavior

We are looking to send a push asking current users to fill out a feedback form (so we would like the push to redirect to a webpage)

### PR DESCRIPTION
**Pull request by jeanregisser**
### Description

This PR adds support for push notifications with an "open url" semantic.
When the `ou` (Open Url) data field is set, it redirects the user to that url.

### Other changes

- Fix notification not handled when app is launched because of said notification
- Fix notification body not displayed
- Fix banner title color
- `AlertBanner`/`showMessage` now supports an action parameter
- Add test for notifications handling

### Tested

**Example marketing notification**

**Payment received (before/after)**

**Banner (before/after)**

### Related issues

- Fixes #5325

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/mobile/src/firebase/firebase.ts [hunk 1]
```diff
@@ -138,14 +138,13 @@ export function* initializeCloudMessaging(app: ReactNativeFirebase.Module, addre
...
-  const initialNotification = yield call([app.messaging(), 'getInitialNotification'])
+  // Manual type checking because yield calls can't infer return type yet :'(
+  const initialNotification: Awaited<ReturnType<
+    FirebaseMessagingTypes.Module['getInitialNotification']
+  >> = yield call([app.messaging(), 'getInitialNotification'])
   if (initialNotification) {
-    Logger.info(TAG, 'App opened fresh via a notification')
-    yield call(
-      handleNotification,
-      initialNotification.notification,
```
Comment (jeanregisser) [inline packages/mobile/src/firebase/firebase.ts:146]: We were bit again by `redux-saga` types not being inferred correctly. 
Here leading to the initial notification to not be handled correctly.

**Review by gnardini — APPROVED** [review]
Looks great!! Just wondering, does Ankit test push notifications? I imagine he might have a hard time receiving one, right? What's the easiest way to receive one for testing?

**Comment by jeanregisser** [comment]
Yes he tests notifications between 2 devices and has reported when they were missing in the past.

**Comment by Lss-Ankit** [comment]
Hi @gnardini Verified task using latest Android play store internal build v1.3.0(1004294316) & Test Flight build v1.3.0(27) and found the following: 

- " Payment Received You've received # celo dollars"  in-app notification is shown when user A is received amount from user B.
We are not getting "Hey Tap this to send your feedback" & "Alfajore A friendly reminder you're ...."  in app notification can you please let us know criteria to get the the in -app notification. 

Note: Previously "Alfajore A friendly reminder you're ...."  in app notification is shown when user fresh sign up but now we are not getting even after fresh sign up in the app. 

Verified devices: 
- iPhone SE 2 (13.5) , Google Pixel Xl (8.1) 
CC: @jeanregisser 
Thanks!

**Comment by jeanregisser** [comment]
Thanks @Lss-Ankit, all this is expected:
- the example push in the screenshot (Hey Tap this...) is something that can be achieved when manually sending push notifications for marketing purpose.
- the alfajores reminder banner is only present on builds connected to the alfajores network.

**Comment by Lss-Ankit** [comment]
Thanks @jeanregisser  for the update. 

So <span>#</span>1, is there any chance that we can get the  "Hey Tap this to send your feedback" notification for testing purpose? 
<span>#</span>2 yes correct , current build is pointing out to Mainnet so we will not get, do we need to test this on alfajore network if yes can you please provide build details. 
Thanks!

**Comment by jeanregisser** [comment]
@Lss-Ankit ping me tomorrow and I can send you such push notification.

And no need to test the alfajores banner.

Thanks

**Comment by Lss-Ankit** [comment]
Thanks @jeanregisser for the co-ordination. 
We have verified "Hey <username> Tap this to send your feedback" push notification using Test flight build v1.3.0(27) and Android play store internal build v1.3.0(1004294316) and fond below new issue. 

**Verified scenarios:**
1. Verify user get the  "Hey <username> Tap this to send your feedback" In-app notification and user is able to redirect to celo.org by tapping on in-app notification. - **Pass**
 
2. Verify user get the  "Hey <username> Tap this to send your feedback"  Push notification and user is able to redirect to celo.org by tapping on push notification if app is running in the background. - **Pass**

3. Verify user get the  "Hey <username> Tap this to send your feedback"  Push notification and user is able to redirect to celo.org by tapping on push notification if app is not running in the background. - **Pass for IOS,** but **Failed for Android**

**New Issue:** https://github.com/celo-org/celo-monorepo/issues/5592

CC: @gnardini

**Pull request merged**